resource_estimation/Error_Budget_Optimization/generate_data.py

```diff
@@ -0,0 +1,139 @@
+
+    running_metric = None
+
+    running_optimal_parameters = {}
+    logical_cache = []
+    t_cache = []
+    rotation_cache = []
+
+    for _i in range(num_iterations):
+        parameters.error_budget = ErrorBudgetPartition()
+        logical_budget_random = np.random.uniform(0, 1)
+        t_budget_random = np.random.uniform(0, 1)
+        rotation_budget_random = np.random.uniform(0, 1)
+        budget_sum = logical_budget_random + t_budget_random + rotation_budget_random
+
+        logical_budget = (logical_budget_random / budget_sum) * total_budget
+        t_budget = (t_budget_random / budget_sum) * total_budget
+        rotation_budget = (rotation_budget_random / budget_sum) * total_budget
+
+        parameters.error_budget.logical = logical_budget
+        parameters.error_budget.t_states = t_budget
+        parameters.error_budget.rotations = rotation_budget
+
+        result = counts.estimate(params=parameters)
+        default_result = counts.estimate()
+
+        physicalqubits = result["physicalCounts"]["physicalQubits"]
+        runtime = result["physicalCounts"]["runtime"]
+
+        default_metric = default_runtime * default_physicalqubits
+        current_metric = runtime * physicalqubits
+
+        if running_metric is None:
+            running_metric = current_metric
+            logical_cache = logical_budget
+            t_cache = t_budget
+            rotation_cache = rotation_budget
+
+        if current_metric < running_metric:
+            running_metric = current_metric
+            logical_cache = logical_budget
+            t_cache = t_budget
+            rotation_cache = rotation_budget
+
+    running_optimal_parameters["logical_budget"] = logical_cache
+    running_optimal_parameters["t_budget"] = t_cache
+    running_optimal_parameters["rotation_budget"] = rotation_cache
+
+    return list(running_optimal_parameters.values()), running_metric, default_metric
+
+
+def generate_data(total_error_budget, counts, path="MQTBench"):
+    """
+    Generates a dataset consisting of logical counts of quantum circuits and respective optimized error budgets.
+
+    This function searches for QASM files in the "MQTBench" directory, loads each circuit,
+    estimates its logical counts, and computes optimized error budget partitions using random sampling.
+    For each circuit, it collects relevant counts and the optimal error budget distribution,
+    appending the results to a list.
+
+    Args:
+        total_error_budget: The total error budget to be distributed among error types.
+        counts: LogicalCounts object or dictionary containing logical counts for estimation.
+
+    Returns:
+        A list of lists, where each inner list contains circuit-specific counts and the corresponding
+        optimized error budget partition.
+    """
+    qasm_files = [Path(root) / file for root, _, files in os.walk(path) for file in files if file.endswith(".qasm")]
+    qasm_files = sorted(qasm_files)
+    print(qasm_files[-1])
```

Comment:
This very much seems like a stray debug line.
```suggestion
```
Print statements should typically not be part of library code.